analytic_tag_default/models/account_analytic_default.py

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+
+class AccountAnalyticDefault(models.Model):
+
+    _inherit = "account.analytic.default"
+
+    tag_ids = fields.Many2many('account.analytic.tag', string="Analytic tags")
+
+    @api.one
+    @api.constrains('analytic_id', 'tag_ids')
+    def _check_required(self):
+        # As at least one field of analytic_id and tag_ids is required we throw
+        # a validation error should both fields be empty
+        if not self.analytic_id and not self.tag_ids:
+            raise ValidationError(_(
+                'At least one Analytic Account or Analytic Tag has to be '
+                'defined.'))
```

Comment:
If you move the groups restriction on the definition of the inherited view (see another comment), you don't need to do this check if user don't have rights on tags.